tools/runners/run-app.js

```diff
@@ -846,28 +790,17 @@ _.extend(AppRunner.prototype, {
     }
 
     function pauseClient(arch) {
-      return appProcess.proc.sendMessage("webapp", {
-        method: "pauseClient",
-        args: [arch],
-      });
+      return appProcess.proc.sendMessage("webapp-pause-client", { arch });
     }
 
     async function refreshClient(arch) {
       if (typeof arch === "string") {
         // This message will reload the client program and unpause it.
-        await appProcess.proc.sendMessage("webapp", {
-          method: "generateClientProgram",
-          args: [arch],
-        });
+        await appProcess.proc.sendMessage("webapp-reload-client", { arch });
       }
-
-      await appProcess.proc.sendMessage("autoupdate", {
-        refresh: "client"
-      }).catch(error => {
-        if (error.message !== "Package autoupdate not installed") {
-          throw error;
-        }
-      });
+      // If arch is not a string, the receiver of this message should
+      // assume all clients need to be refreshed.
+      await appProcess.proc.sendMessage("client-refresh");
```

Comment:
If the `autoupdate` package is not installed, no message will be delivered to that package, so no error needs to be caught here.